### PR TITLE
Update CHANGELOG.md for 2.3.2

### PR DESCRIPTION
**Pull request by daschuer**
This updates the CHANGELOG for a 2.3.2 release.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,14 @@
 ## [2.3.2](https://launchpad.net/mixxx/+milestone/2.3.2) (Unreleased)
 
 * Improve robustness of file type detection by considering the actual MIME type of the content. [lp:1445885](https://bugs.launchpad.net/mixxx/+bug/1445885) [#4356](https://github.com/mixxxdj/mixxx/pull/4356) [#4357](https://github.com/mixxxdj/mixxx/pull/4357)
+* Fix/Improve Appstream metainfo
```
Comment (Holzhaus) [inline CHANGELOG.md:6]: This should go into a packaging section because it affects package maintainers.

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,14 @@
...
+  [#4344](https://github.com/mixxxdj/mixxx/pull/4344)
+  [#4346](https://github.com/mixxxdj/mixxx/pull/4346)
+  [#4349](https://github.com/mixxxdj/mixxx/pull/4349)
+* Playlist: enable sorting by color [#4352](https://github.com/mixxxdj/mixxx/pull/4352) [lp:1945976](https://bugs.launchpad.net/mixxx/+bug/1945976)
+* CMake: Adjust the download directory and name of external dependencies [#4511](https://github.com/mixxxdj/mixxx/pull/4511)
+* Fix clang compiler warnings [#4551](https://github.com/mixxxdj/mixxx/pull/4551) [#4553](https://github.com/mixxxdj/mixxx/pull/4553)
+* ClangFormat: Enable c++17 formatting [#4557](https://github.com/mixxxdj/mixxx/pull/4557)
```
Comment (Holzhaus) [inline CHANGELOG.md:13]: I think that does not belong into the changelog because it's irrelevant to the user.

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,14 @@
 ## [2.3.2](https://launchpad.net/mixxx/+milestone/2.3.2) (Unreleased)
 
 * Improve robustness of file type detection by considering the actual MIME type of the content. [lp:1445885](https://bugs.launchpad.net/mixxx/+bug/1445885) [#4356](https://github.com/mixxxdj/mixxx/pull/4356) [#4357](https://github.com/mixxxdj/mixxx/pull/4357)
+* Fix/Improve Appstream metainfo
+  [#4344](https://github.com/mixxxdj/mixxx/pull/4344)
+  [#4346](https://github.com/mixxxdj/mixxx/pull/4346)
+  [#4349](https://github.com/mixxxdj/mixxx/pull/4349)
+* Playlist: enable sorting by color [#4352](https://github.com/mixxxdj/mixxx/pull/4352) [lp:1945976](https://bugs.launchpad.net/mixxx/+bug/1945976)
+* CMake: Adjust the download directory and name of external dependencies [#4511](https://github.com/mixxxdj/mixxx/pull/4511)
+* Fix clang compiler warnings [#4551](https://github.com/mixxxdj/mixxx/pull/4551) [#4553](https://github.com/mixxxdj/mixxx/pull/4553)
```
Comment (Holzhaus) [inline CHANGELOG.md:12]: I don't think this needs to be mentioned in the changelog.

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,14 @@
 ## [2.3.2](https://launchpad.net/mixxx/+milestone/2.3.2) (Unreleased)
 
 * Improve robustness of file type detection by considering the actual MIME type of the content. [lp:1445885](https://bugs.launchpad.net/mixxx/+bug/1445885) [#4356](https://github.com/mixxxdj/mixxx/pull/4356) [#4357](https://github.com/mixxxdj/mixxx/pull/4357)
+* Fix/Improve Appstream metainfo
+  [#4344](https://github.com/mixxxdj/mixxx/pull/4344)
+  [#4346](https://github.com/mixxxdj/mixxx/pull/4346)
+  [#4349](https://github.com/mixxxdj/mixxx/pull/4349)
+* Playlist: enable sorting by color [#4352](https://github.com/mixxxdj/mixxx/pull/4352) [lp:1945976](https://bugs.launchpad.net/mixxx/+bug/1945976)
```
Comment (Holzhaus) [inline CHANGELOG.md:10]:
```suggestion
* Playlist: Enable sorting by color [#4352](https://github.com/mixxxdj/mixxx/pull/4352) [lp:1945976](https://bugs.launchpad.net/mixxx/+bug/1945976)
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,6 +3,14 @@
...
+* CMake: Adjust the download directory and name of external dependencies [#4511](https://github.com/mixxxdj/mixxx/pull/4511)
```
Comment (Holzhaus) [inline CHANGELOG.md:11]: @uklotzde is this relevant to package maintainers where build servers mustn't access the internet?

If so, this should go into a packaging section. If not, this doesn't need to be mentioned at all.

**Comment by daschuer** [comment]
Done

**Comment by Swiftb0y** [comment]
Why are we adding PRs to the 2.3.1 and older changelog sections retroactively?

**Comment by daschuer** [comment]
> Why are we adding PRs to the 2.3.1 and older changelog sections retroactively?

I have added https://github.com/mixxxdj/mixxx/pull/4070, because it was int my Todo list for 2.4 and it turns out it was already part of 2.3.1  That's the only reason. I actually don't mind to keep or remove it.

**Comment by Swiftb0y** [comment]
Thanks for the explanation. Going back and adding PRs we forgot to put into the changelog is fine IMO.

**Comment by ronso0** [comment]
btw: #4531 and #4539 are ready for 2.3.2, only need a final LGTM

**Pull request merged**